File: UL/System/Console.cs
namespace ul
{
    namespace System
    {
        public class Console
        {
            public extern static void Write(string value);

            public static void Write(object value)
            {
                Write(value.ToString());
            }
            public static void Write(char value)
            {
                Write(value.ToString());
            }

            public static void Write(bool value)
            {
                Write(value.ToString());
            }

            public static void Write(int value)
            {
                Write(value.ToString());
            }
            public static void Write(long value)
            {
                Write(value.ToString());
            }

            public static void Write(float value)
            {
                Write(value.ToString());
            }

            public static void Write(double value)
            {
                Write(value.ToString());
            }

            public static void Write(byte value)
            {
                Write(value.ToString());
            }

            public static void WriteLine()
            {
                Write("\r\n");
            }

            public static void WriteLine(char value)
            {
                Write(value.ToString());
                WriteLine();
            }

            public static void WriteLine(bool value)
            {
                Write(value.ToString());
                WriteLine();
            }

            public static void WriteLine(int value)
            {
                Write(value.ToString());
                WriteLine();
            }
            public static void WriteLine(long value)
            {
                Write(value.ToString());
                WriteLine();
            }

            public static void WriteLine(float value)
            {
                Write(value.ToString());
                WriteLine();
            }

            public static void WriteLine(double value)
            {
                Write(value.ToString());
                WriteLine();
            }

            public static void WriteLine(byte value)
            {
                Write(value.ToString());
                WriteLine();
            }
            public static void WriteLine(string value)
            {
                Write(value);
                WriteLine();
            }
            public static void WriteLine(object value)
            {
                Write(value.ToString());
                WriteLine();
            }
        }
    }
}

File: UL/CppApp/System/Console.cpp
#include "stdafx.h"
#include "System\Console.h"
#include "System\Object.h"
#include "System\String.h"
#include "System\Char.h"
#include "System\Boolean.h"
#include "System\Int32.h"
#include "System\Int64.h"
#include "System\Single.h"
#include "System\Double.h"
#include "System\Byte.h"
#include "Console_ExtCpp.h"
void System::Console::Write(Ref<System::Object>  value)
{
	Write(value->ToString());
}
void System::Console::Write(System::Char  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Boolean  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Int32  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Int64  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Single  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Double  value)
{
	Write(value.ToString());
}
void System::Console::Write(System::Byte  value)
{
	Write(value.ToString());
}
void System::Console::WriteLine()
{
	Write(Ref<System::String>(new System::String(_T("\r\n"))));
}
void System::Console::WriteLine(System::Char  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Boolean  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Int32  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Int64  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Single  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Double  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(System::Byte  value)
{
	Write(value.ToString());
	WriteLine();
}
void System::Console::WriteLine(Ref<System::String>  value)
{
	Write(value);
	WriteLine();
}
void System::Console::WriteLine(Ref<System::Object>  value)
{
	Write(value->ToString());
	WriteLine();
}
